src/query/functions/temporal/rate.go

```diff
@@ -108,7 +109,11 @@ func NewRateOp(args []interface{}, optype string) (transform.Params, error) {
 	return newBaseOp(duration, optype, r)
 }
 
-type rateFn func(ts.Datapoints, bool, bool, int64, int64, time.Duration) float64
+type rateFn func(
+	datapoints ts.Datapoints,
+	isRate bool, isCounter bool,
+	rangeStart xtime.UnixNano, rangeEnd xtime.UnixNano,
```

Comment:
nit: Can leave type out if args next to each other are same type. e.g. `isRate, isCounter bool, rangeStart, rangeEnd xtime.UnixNano`

Reply:
Ah yeah, so I generally prefer to do a type per arg, just find it easier to grok on a quick glance; I'll split by line here but happy to change to the other approach if that's the preferred style